src/ImageRunModal.jsx

```diff
@@ -32,135 +38,177 @@ const units = {
-    (
-        <>
-            <InputGroup className='ct-input-group-spacer-sm' id={id}>
-                <TextInput aria-label={_("IP (optional)")}
-                           type='text'
-                           placeholder={_("IP (optional)")}
-                           value={item.IP || ''}
-                           onChange={value => onChange(idx, 'IP', value)} />
-                <TextInput aria-label={_("Host port (optional)")}
-                           type='number'
-                           step={1}
-                           min={1}
-                           max={65535}
-                           placeholder={_("Host port (optional)")}
-                           value={item.hostPort || ''}
-                           onChange={value => onChange(idx, 'hostPort', value)} />
-                <Button variant='secondary'
-                        className={"btn-close" + (idx === 0 && !item.IP && !item.hostPort && !item.containerPort ? ' invisible' : '')}
-                        isSmall
-                        aria-label={_("Remove item")}
-                        icon={<CloseIcon />}
-                        onClick={() => removeitem(idx)} />
-                <Button variant='secondary' className="btn-add" onClick={additem} aria-label={_("Add item")} icon={<PlusIcon />} />
-            </InputGroup>
-            <InputGroup className='ct-input-group-spacer-sm'>
-                <TextInput aria-label={_("Container port")}
-                           type='number'
-                           step={1}
-                           min={1}
-                           max={65535}
-                           placeholder={_("Container port")}
-                           value={item.containerPort || ''}
-                           onChange={value => onChange(idx, 'containerPort', value)} />
-                <FormSelect className='pf-c-form-control container-port-protocol'
-                            aria-label={_("Protocol")}
-                            value={item.protocol}
-                            onChange={value => onChange(idx, 'protocol', value)}>
-                    <FormSelectOption value='tcp' key='tcp' label={_("TCP")} />
-                    <FormSelectOption value='udp' key='udp' label={_("UDP")} />
-                </FormSelect>
-            </InputGroup>
-        </>
-    );
+const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount }) => {
+    if (itemCount) {
+        return (
+            <Grid hasGutter id={id}>
+                <FormGroup className="pf-m-4-col-on-sm"
+                    label={_("IP address")}
+                    fieldId={id + "-ip-address"}
+                    labelIcon={
+                        <Popover aria-label={_("IP address help")}
+                            enableFlip
+                            bodyContent={_("If host IP is set to 0.0.0.0 or not set at all, the port will be bound on all IPs on the host.")}>
```

Comment:
Should the default value be 0.0.0.0 then?

Reply:
I don't like the idea of prefilling with default values.

Reply:
Ok :+1: